### DISCUSSION TITLE
postinstall vs posttrans

### DISCUSSION THREAD

**Discussion by suntong** [goreleaser/nfpm#418] [Q&A] [answered]
Are `postinstall` and `posttrans` different stages?

- I see `The posttrans script runs after all RPM package transactions / stages` for `rpm`, is such stage available for `deb` as well?
- what does `postinstall` stage mean then? 

I put 

`postinstall: chmod 644 /etc/myapp-*`

but got error when ***building*** my `deb` package. 

Wouldn't the `postinstall` scripts ran when after the package is installed?
pls help. thx

**Comment by djgilcrease** [comment, marked as answer]
deb does not have a post transaction stage.

What is the error you are getting when building the package?

**Reply by suntong** [reply]
`open chmod 644 /etc/myapp-*: no such file or directory`

maybe I should use a script instead of directly using command?

**Reply by djgilcrease** [reply]
yes the `postinstall` field should point to a file not the actual script code
